### ISSUE TITLE
A real Home Page

### ISSUE THREAD

**Issue by lu4p** [libre-tube/LibreTube#918]
### Describe your suggested feature

I have an idea for something more like a real YouTube home feed.

I think it should include:

1. Subscription videos which are not viewed yet from the past week.

2. Videos related to 10 recently watched videos:
- could be a lower initial number depending on performance 
- number of videos recommend from increases when the user reaches the end of the feed
- some filtering should apply here, only: 
    - x videos from the same channel (reverse also applies only use up to 3 videos from the same channel to recommend from)
    - recommend a few related videos from a video
    - recommend from videos which are at least x% watched through
   -  videos which have not been watched before

3. Videos from a watch later playlist:
    - this would require a default "watch later" playlist which cannot be deleted
    - "Watch later" would also need to be in the long press thumbnail menu
   - "Watch later" is a special playlist it gets it's own space separate from real playlists everywhere

There should also be an option to block a specific video or creator from the home feed, these should be in the long press menu "Not interested" and "Don't recommend channel". These blocks might be reversible in the settings but probably only to unblock entire creators.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
There already were discussions about something like that but I'm not in favor of it.
1.Since related videos and stuff like that are also kind of an algorithm, this could users force to become addicted to watching videos on their suggestions feed and makes it harder to stop watching.
2. This would force us to fetch the video information of each video one by one which would result in a very low performance (it would probably take tens of seconds minimum, rather more).
3. It would require storing a lot of user data inside the app, the amount of storage for it would probably rise a lot. I don't think tracking too much user data is a good thing although it's still only locally. The benefits of projects like that in my opinion that you don't spread data and also feel like you really own control. I don't want it to become like person related advertising.
4. About a different a home feed: There was an idea to make a trending page which uses anonymous sent stats of the instance. That wouldn't be personally related but it might still be better than the current YouTube trending tab.
